### ISSUE TITLE
Why did 731bca35b (remove __repr__ from API classes) fix the tests?

### ISSUE THREAD

**Issue by 153957** [HiSPARC/sapphire#153]
I did not full investigate what caused the tests to fail before 731bca35b, but removing the newly added `__repr__` methods fixed the tests. The failure: https://travis-ci.org/HiSPARC/sapphire/builds/142904800 .

Of the top of my head I could think that it may be caused by a changed test order execution...
Perhaps a very barebones `__repr__` can be added to see if that also causes the error..

**Comment by tomkooij** [comment]
at 02afabb :
`$ python sapphire/tests/test_api.py` **fails** in `StationTests.test_station_timing_offsets` 
but:
`$python sapphire/tests/test_api.py StationTests.test_station_timing_offsets` **passes**.

So it is indeed a test order thing. Something in `test_api.py` breaks the test. Should be easy enough to bisect using: `python sapphire/tests/test_api.py -v`

**Comment by 153957** [comment]
I tried simply adding:
```python
def __repr__(self):
    return ''
```
Which caused some tests to fail. If it call the test directly I got a different error, but still an unexpected error.

```python
# python test_api.py StationTests.test_station_timing_offsets
F
======================================================================
FAIL: test_station_timing_offsets (__main__.StationTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_api.py", line 553, in test_station_timing_offsets
    self.assertRaises(Exception, self.station.station_timing_offsets, STATION)
AssertionError: Exception not raised
```

Adding a `__str__` method instead of `__repr__` does not cause issues.
I also tried splitting the tests into separate testCases, but that also did not help.
This requires some additional debugging.

**Comment by tomkooij** [comment]
It's probably `memoize`

**Comment by 153957** [comment]
I stole a modified version of memoize from http://stackoverflow.com/a/29954160/1033535
Which I also modified to fix the `key` used for the cache.
This stores the cache on `self`. This seems to have fixed the problem, and all tests pass.

**Comment by 153957** [comment]
Fixed by b5ea49c
